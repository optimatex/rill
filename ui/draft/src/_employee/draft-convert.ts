export { convertToHTML, convertFromHTML } from 'draft-convert';
