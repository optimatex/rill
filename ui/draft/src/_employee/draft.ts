export { Editor, EditorState, convertToRaw } from 'draft-js';
