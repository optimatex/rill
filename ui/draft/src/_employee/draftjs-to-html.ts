export { default as draftToHtml } from 'draftjs-to-html';
