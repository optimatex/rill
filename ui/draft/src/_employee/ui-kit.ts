export { Button, ButtonActive } from '@rill/ui-kit';
