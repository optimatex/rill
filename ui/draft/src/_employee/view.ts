export {
  React,
  useState,
  useCallback,
  css,
  styled,
  createGlobalStyle,
} from '@rill/view';
