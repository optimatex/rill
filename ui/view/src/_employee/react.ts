import * as React from 'react';

import {
  useCallback,
  useContext,
  useEffect,
  useMemo,
  useReducer,
  useRef,
  useState,
  createContext,
} from 'react';

export {
  React,
  useCallback,
  useContext,
  useEffect,
  useMemo,
  useReducer,
  useRef,
  useState,
  createContext,
};
