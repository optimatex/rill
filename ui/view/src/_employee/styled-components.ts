export {
  css,
  createGlobalStyle,
  ThemeProvider,
  ThemeContext,
} from 'styled-components';

export { default as styled } from 'styled-components';
