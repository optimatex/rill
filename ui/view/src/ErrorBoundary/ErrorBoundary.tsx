import { React } from '_employee/react';

import { PlainRender, IContainerProps } from '@types';

type UnknownError = unknown;

interface IProps extends IContainerProps {
  renderFallback: PlainRender;
  scope: string;
  onError: (scope: string, error: UnknownError) => void;
}

interface IState {
  error: UnknownError;
}

class ErrorBoundary extends React.Component<IProps, IState> {
  public state = {
    error: false,
  };

  public handleRestore = () => {
    this.setState({
      error: null,
    });
  };

  static getDerivedStateFromError(error: UnknownError) {
    return {
      error,
    };
  }

  public componentDidCatch(error: UnknownError) {
    this.props.onError(this.props.scope, error);
  }

  public render() {
    return this.state.error ? this.props.renderFallback() : this.props.children;
  }
}

export default ErrorBoundary;
