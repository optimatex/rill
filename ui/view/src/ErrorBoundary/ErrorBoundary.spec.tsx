import { React } from '_employee/react';
import { render, renderNumber, ProblemChild, fireEvent } from 'test';
import ErrorBoundary from '.';

const text = 'Some content';
const errorText = 'some error';

const genProps = () => ({
  renderFallback: renderNumber,
  onError: jest.fn(),
});

const draw = (props) =>
  render(
    <ErrorBoundary {...props}>
      <ProblemChild errorText={errorText} properText={text} />
    </ErrorBoundary>,
  );

describe('ErrorBoundary', () => {
  it('should render proper children in case no error occured', () => {
    const props = genProps();

    const { getByText } = draw(props);
    const el = getByText(text);

    expect(el).toMatchSnapshot();
  });

  it('should render error card in case an error occured', () => {
    const props = genProps();

    const { getByText, getByRole } = draw(props);
    const el = getByRole('button');

    try {
      expect(el).toMatchSnapshot();

      // now trigger error
      // fireEvent.click(el);
    } catch (error) {
      const withError = getByText(errorText);

      // eslint-disable-next-line jest/no-conditional-expect
      // expect(withError).toMatchSnapshot();
    }
  });
});
