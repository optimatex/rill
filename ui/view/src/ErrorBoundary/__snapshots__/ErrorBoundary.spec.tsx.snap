// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorBoundary should render error card in case an error occured 1`] = `
<button>
  Some content
</button>
`;

exports[`ErrorBoundary should render proper children in case no error occured 1`] = `
<button>
  Some content
</button>
`;
